src/radio-group/index.tsx

```diff
@@ -2,16 +2,31 @@
 // SPDX-License-Identifier: Apache-2.0
 import React from 'react';
 
+import { getAnalyticsMetadataAttribute } from '@cloudscape-design/component-toolkit/internal/analytics-metadata';
+
 import useBaseComponent from '../internal/hooks/use-base-component';
 import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { GeneratedAnalyticsMetadataRadioGroupComponent } from './analytics-metadata/interfaces';
 import { RadioGroupProps } from './interfaces';
 import InternalRadioGroup from './internal';
 
 export { RadioGroupProps };
 
 const RadioGroup = React.forwardRef((props: RadioGroupProps, ref: React.Ref<RadioGroupProps.Ref>) => {
   const baseComponentProps = useBaseComponent('RadioGroup');
-  return <InternalRadioGroup ref={ref} {...props} {...baseComponentProps} />;
+  return (
+    <InternalRadioGroup
+      ref={ref}
+      {...props}
+      {...baseComponentProps}
+      {...getAnalyticsMetadataAttribute({
```

Comment:
Why do we use __injectAnalyticsComponentMetadata for checkbox but pass metadata directly for radio?

Reply:
because in checkbox the `component` information needs to be injected in the same element as the event-specific information. In radio group the 2 can be injected in separate elements